precompiles/staking/types.go

```diff
@@ -49,6 +56,110 @@ type EventCancelUnbonding struct {
 	CreationHeight   *big.Int
 }
 
+// Description defines a validator description.
+type Description struct {
+	Moniker         string "json:\"moniker\""
+	Identity        string "json:\"identity\""
+	Website         string "json:\"website\""
+	SecurityContact string "json:\"securityContact\""
+	Details         string "json:\"details\""
+}
+
+// Commission defines a validator commission.
+// since solidity does not support decimals, after passing in the big int, convert the big int into a decimal with a precision of 18
+type Commission struct {
+	Rate          *big.Int "json:\"rate\""
+	MaxRate       *big.Int "json:\"maxRate\""
+	MaxChangeRate *big.Int "json:\"maxChangeRate\""
+}
+
+// NewMsgCreateValidator creates a new MsgCreateValidator instance and does sanity checks
+// on the given arguments before populating the message.
+func NewMsgCreateValidator(args []interface{}, denom string) (*stakingtypes.MsgCreateValidator, common.Address, error) {
+	if len(args) != 7 {
+		return nil, common.Address{}, fmt.Errorf(cmn.ErrInvalidNumberOfArgs, 7, len(args))
+	}
+
+	descriptionJson, err := json.Marshal(args[0])
+	if err != nil {
+		return nil, common.Address{}, fmt.Errorf("failed to marshal description to json: %v", err)
+	}
+	var description Description
+	if err := json.Unmarshal(descriptionJson, &description); err != nil {
+		return nil, common.Address{}, fmt.Errorf("failed to unmarshal json to description: %v", err)
+	}
```

Comment:
@Vvaradinov are we doing this with the other methods? this is tech debt and costly. We shouldn't have to encode and decode the values

Reply:
Yes, I also find this method unsuccinct. If there is a better way, please provide it to me.

Reply:
@luchenqun we would need to modify the ABI unpack function for integers to do this easily: 

https://github.com/ethereum/go-ethereum/blob/master/accounts/abi/unpack.go#L38

Reply:
another method has been used to decode the description, please review